src/cpp/rtps/builtin/discovery/participant/PDPServerListener2.cpp

```diff
@@ -124,25 +124,58 @@ void PDPServerListener2::onNewCacheChangeAdded(
                     pdp_server()->getRTPSParticipant()->network_factory(),
                     pdp_server()->getRTPSParticipant()->has_shm_transport()))
         {
-            // // Key should not match instance handle
-            // if (change->instanceHandle == temp_participant_data_.m_key)
-            // {
-            //     logInfo(RTPS_PDP_LISTENER, "Malformed PDP payload received, ignoring: " << change->instanceHandle);
-            //     return;
-            // }
-
-            // Check whether the participant is a CLIENT or a SERVER
-            bool is_client = true;
             fastrtps::ParameterPropertyList_t properties = temp_participant_data_.m_properties;
-            for (auto property : properties)
+
+            /* Check DS_VERSION */
+            auto ds_version = std::find_if(
+                properties.begin(),
+                properties.end(),
+                [](const dds::ParameterProperty_t& property)
+                {
+                    return property.first() == "DS_VERSION";
+                });
+
+            if (ds_version != properties.end())
+            {
+                if (std::stof(ds_version->second()) < 2.0)
+                {
+                    logError(RTPS_PDP_LISTENER, "Minimum DS_VERSION is 2.0, found: " << ds_version->second());
```

Comment:
This will break compatibility backwards with the old server.
This may be a possibility in future releases when backup feature is implemented, but right now the backup feature uses the old version of DS_VERSION (not even set) and it will make impossible the use of backup feature.
It would not be a problem to treat a new Participant of old server versions as a standar entity, till backup is implemented. This new entity without version should be a SERVER, because the client is shared between versions, and so we can deduce that a client will always be version 2.

Reply:
Done